app/services/catalog/share_info.rb

```diff
@@ -0,0 +1,31 @@
+module Catalog
+  class ShareInfo
+    require 'rbac-api-client'
+    attr_reader :result
+
+    def initialize(options)
+      @object = options[:object]
+    end
+
+    def process
+      group_permissions = {}
+      @object.access_control_entries.each do |ace|
+        group_permissions[ace.group_uuid] = group_permissions.fetch(ace.group_uuid, []) << ace.permission
+      end
+      @result = group_permissions.each_with_object([]) do |(k, v), memo|
+        memo << { :group_name => group_names[k], :group_uuid => k, :permissions => v }
```

Comment:
We could have stale group.uuid in our database which has been deleted from the RBAC Service, should we skip those here and leave debris in our database. The group existed when the sharing was created but after that the user deleted the group and we have it in the catalog DB as left over.